dir.props

```diff
@@ -117,6 +117,27 @@
   </PropertyGroup>
 
   <Import Project="$(BuildConfigurationImportFile)" Condition="Exists('$(BuildConfigurationImportFile)')" />
+  <PropertyGroup>
+    <_runtimeOSVersionIndex>$(RuntimeOS.IndexOfAny(".-0123456789"))</_runtimeOSVersionIndex>
+    <_runtimeOSFamily Condition="'$(_runtimeOSVersionIndex)' != '-1'">$(RuntimeOS.SubString(0, $(_runtimeOSVersionIndex)))</_runtimeOSFamily>
+    <_portableRID>linux</_portableRID>
+    <_portableRID Condition="'$(_runtimeOSFamily)' == 'win'">win</_portableRID>
+    <_portableRID Condition="'$(_runtimeOSFamily)' == 'osx'">osx</_portableRID>
+    <!-- support cross-targeting by choosing a RID to restore when running on a different machine that what we're build for -->
+    <_portableRID Condition="'$(OSGroup)' == 'Unix' AND '$(_runtimeOSFamily)' != 'osx'">linux</_portableRID>
+    <_portableRID Condition="'$(OSGroup)' == 'Windows_NT'">win</_portableRID>
+
+    <_packageRID/>
+    <_packageRID Condition="'$(PortableBuild)' == 'true'">$(_portableRID)-$(ArchGroup)</_packageRID>
+    <_packageRID Condition="'$(TargetGroup)' == 'uap'">win10-$(ArchGroup)</_packageRID>
```

Comment:
Similarly, this logic is specific to the UAP meta-package projects so it may make more sense just to keep it in those projects. /cc @joperezr